src/LanguageServerProtocol/LanguageServerProtocol.fs

```diff
@@ -2528,7 +2585,18 @@ module Server =
             | MessageType.Request ->
                 let request = JsonConvert.DeserializeObject<JsonRpc.Request>(messageString, jsonSettings)
                 async {
-                  let! result = handleRequest requestHandlings request lspServer
+                  use cts = new System.Threading.CancellationTokenSource()
+                  use _ = cts.Token.Register(fun _ -> responseAgent.Post(RequestCompleted request.Id))
```

Comment:
now that I think of it, this is wrong. we don't want these events to fire when the token is cancelled, but rather we want to trigger this post when the scope of this ends. Probably better to use an inline Idisposable....